packages/govuk-frontend-review/src/views/examples/javascript-errors/index.njk

```diff
@@ -0,0 +1,40 @@
+{% extends "layouts/layout.njk" %}
+
+{% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+{% from "govuk/components/character-count/macro.njk" import govukCharacterCount %}
+
+{% block beforeContent %}
+  {{ govukBackLink({
+    href: "/"
+  }) }}
+{% endblock %}
+
+{% block content %}
+  <h1 class="govuk-heading-l">JavaScript errors</h1>
+  <p class="govuk-body-lead">
+    Open your <a class="govuk-link" href="https://developer.mozilla.org/en-US/docs/Web/API/console#see_also">brower's console</a>
+    to witness the error thrown in browsers where GOV.UK Frontend is not supported.
+  </p>
+  {{ govukCharacterCount({
+    label: {
+      text: 'Any other comment'
+    },
+    id: 'some-content',
+    name: 'some-content'
+  })}}
+{% endblock %}
+
+{% block bodyEnd %}
+
+  <script type="module" src="/javascripts/govuk-frontend.min.js"></script>
+
+  <script type="module">
+    import { Accordion } from '/javascripts/govuk-frontend.min.js'
+    // Simulate GOV.UK Frontend not being supported
+    document.body.classList.remove('govuk-frontend-supported')
+
+    // We don't want to use `initAll` as it returns early if govuk-frontend is not supported
+    const $element = document.querySelector('[data-module="govuk-character-count"]')
```

Comment:
Mind if we clarify that "return early" actually "will catch and log errors" in https://github.com/alphagov/govuk-frontend/pull/4067?

```suggestion

    // Simulate element not found
    $element = null

    // Enable `SupportError` by removing GOV.UK Frontend support
    document.body.classList.remove('govuk-frontend-supported')
    
    // Throw `SupportError` by instantiating component directly
    // since calling `initAll()` will catch and log errors only
```

Reply:
At the minute, we still have an early return in #4067 which I _think_ is what we want…?

https://github.com/alphagov/govuk-frontend/blob/e6a32b64550eef033600b4229296ad8c66ddbb49/packages/govuk-frontend/src/govuk/all.mjs#L27-L30

Reply:
Ahh here's me thinking we had a helper for this, but `initAll()` caught + logged it 🤦‍♂️

```js
function checkSupport() {
  if (!isSupported()) {
    throw new SupportError()
  }
}
```

Perhaps we do want `checkSupport()` as a helper so we can log something useful for `initAll()` too?

```js
// Skip initialisation when GOV.UK Frontend is not supported
try {
  checkSupport()
} catch (error) {
  console.error(error)
  return
}
``

Reply:
I [don't think we need to make that early return be triggered by an error being thrown](https://github.com/alphagov/govuk-frontend/pull/4117#issuecomment-1686057627), so I'd keep the comment and `initAll`'s behaviour as is.

Reply:
That's nicely cleared up now in https://github.com/alphagov/govuk-frontend/pull/4117

Other lines clarified that it's the class removal that makes `SupportError` possible

But I've used `$element = null` as no Character Count exists on the page, intentionally

```suggestion
    // Simulate element not found
    $element = null

    // Enable `SupportError` by removing GOV.UK Frontend support
    document.body.classList.remove('govuk-frontend-supported')
    
    // Throw `SupportError` by instantiating component directly
    // since calling `initAll()` logs only without throwing
```

Reply:
Update on its way to describe adding new examples in separate script tags.